lib/ZnapZend/ZFS.pm

```diff
@@ -289,6 +314,15 @@ sub sendRecvSnapshots {
     my $recvOpt = $self->recvu ? '-uF' : '-F';
     my $remote;
     my $mbufferPort;
+    my $dstDataSetExists = $self->dataSetExists($dstDataSet);
+
+    #attemp a creation of the dataset when it doesn't exist
+    $dstDataSetExists = $self->createDataSet($dstDataSet) if $self->autoCreation && !$dstDataSetExists;
+
+    #check if the dstDataSet exist on the destination (maybe after a creation)
+    !$dstDataSetExists
+        and Mojo::Exception->throw('ERROR: dstDataSet does not exist on dest host');
+
```

Comment:
Hmm, this will only throw an exception when the dataset isn't there, so this dataset is being skipped (else it will start the transfer, which throws a lof of errors). The createDataSet is only called when autoCreation is enabled and the dataset isn't there yet. 

I've added some extra information to the exception.